lib/delighted/list_resource.rb

```diff
@@ -0,0 +1,41 @@
+module Delighted
+  class ListResource
+    def initialize(klass, path, opts, client)
+      @class = klass
+      @path = path
+      @opts = opts
+      @client = client
+      @iteration_count = 0
+    end
+
+    def auto_paging_each(opts = {})
+      auto_handle_rate_limits = opts.fetch(:auto_handle_rate_limits, true)
+      loop do
+        begin
+          # Get next (or first) page
+          if @iteration_count == 0
+            data = @client.request_get(@path, { params: @opts })
+          else
+            data = @client.request_get(@next_link, { full_url: true })
+          end
+        rescue Delighted::RateLimitedError => e
+          if auto_handle_rate_limits
+            sleep e.response.headers['Retry-After'].to_i
+            retry
+          else
+            raise
+          end
+        end
+
+        @iteration_count += 1
+        @next_link = data[:response].next_link
+
+        data[:json].map do |attributes|
+          yield Utils.full_const_get(@class).new(attributes)
+        end
+
+        break if @next_link.nil?
```

Comment:
What happens if someone calls `auto_paging_each` a second time accidentally? It might be good to set a `@done` instance variable when we hit the end of the pagination and also check it at the start of `auto_paging_each` to give a good exception for this edge case.

Reply:
Yeah, it won't find a proper `next_link`. Debated adding a `people.reset` method but don't see the need for additional complexity. Adding a `@done` as suggested.